client/components/SingleProduct.js

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { getOneProduct } from '../store/singleProduct';
+
+class SingleProduct extends React.Component {
+	constructor() {
+		super();
+		this.state = { quantity: 1 };
+		this.handleAddToCart = this.handleAddToCart.bind(this);
+		this.handleChange = this.handleChange.bind(this);
+	}
+
+	componentDidMount() {
+		const productId = this.props.match.params.productId;
+		this.props.loadProduct(productId);
+	}
+
+	handleAddToCart(event) {
+		event.preventDefault();
```

Comment:
I would take all of this logic out of here, and move it all into redux. handleAddToCart should just add a piece of state to the redux store for the cart. The local storage setting and retrieving should happen in redux as well, as it is technically considered a side effect